### PR TITLE
Use settings.ini file instead of preinstalled.list.lock

### PR DESCRIPTION
**Pull request by SchnWalter**
Instead of using a lock file that only contains a list of files needed to be downloaded you can use a settings INI file that can contain various other settings.

To add a new application from fdroid to you can add an extra entry in the fdroid_apps section.
```ini
[fdroid_apps]
com.fsck.k9=22001
org.videolan.vlc=latest
```

### CODE REVIEW

**Comment by SchnWalter** [comment]
Updated the settings file... but there was one app in the original lock file that I could not find on f-droid.org

`fennec-35.0.multi.android-arm.apk`

**Comment by patcon** [comment]
that's because it's Firefox, and that's the one remaining apk that f-droid doesn't actually build. (note the apk link doesn't match the normal download link format)
https://f-droid.org/repository/browse/?fdfilter=firefox&fdid=org.mozilla.firefox

But that will be going away at some point in the future, as we're working to get it building. Maybe just a hardcoded exception for now when the app ID is for `org.mozilla.firefox=36.0`? And that wierdness will go away soon :/

**Comment by patcon** [comment]
Reviewing this now fyi :)

File: settings.sample.ini [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+;password_file=/path/to/secret/file
+
+; Preferred snapshot version
+snapshot_version=M12
+
+
+; Installed before those downloaded from the f-droid.org repository.
+[apps]
+FDroid=https://f-droid.org/FDroid.apk
+OrfoxFennec-debug=https://guardianproject.info/builds/OrfoxFennec/latest/OrfoxFennec-debug.apk
```
Comment (patcon) [inline settings.sample.ini]: Not a system app, just needed to download from a non-standard url. as-is, it crashes on load

File: settings.sample.ini [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+;       to be secure.
+[fdroid_apps]
+com.artifex.mupdfdemo=55
+com.csipsimple=2459
+com.fsck.k9=22002
+com.morlunk.mumbleclient=72
+com.projectsexception.myapplist.open=16
+com.xabber.androiddev=123
+de.schildbach.wallet=209
+fr.kwiatkowskiTrack=1
```
Comment (patcon) [inline settings.sample.ini]: Guessing some sed'ing borked this from `fr.kwiatkowski.ApkTrack`

File: settings.sample.ini [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+[apps]
+FDroid=https://f-droid.org/FDroid.apk
+OrfoxFennec-debug=https://guardianproject.info/builds/OrfoxFennec/latest/OrfoxFennec-debug.apk
+
+
+; Contains a list of applications to be installed.
+; If you feel adventurous use "latest" to retrieve the latest released version.
+; NOTE: Versions are locked because we want to be sure that the apps are known
+;       to be secure.
+[fdroid_apps]
```
Comment (patcon) [inline settings.sample.ini]: We lost the official mozilla firefox pkg (we just have the custom, bleeding-edge fdroid build). It just so happens that the one we lost is the only working browser atm (fdroid fennec and orfox are crashing)

File: settings.sample.ini [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+com.projectsexception.myapplist.open=16
+com.xabber.androiddev=123
+de.schildbach.wallet=209
+fr.kwiatkowskiTrack=1
+indrora.atomic=19
+net.osmand.plus=197
+org.coolreader=875
+org.ethack.orwall=35
+org.mariotaku.twidere=98
+org.mozilla.fennec_fdroid=360010
```
Comment (patcon) [inline settings.sample.ini]: crashing on startup. Not sure if it's something we did, or just a bad version on my device...

Comment (patcon) [inline settings.sample.ini]: Ah. I guess i lucked out before when grabbing the top download url via xpath, but seems some apps have different builds for `x86` vs `armeabi-v7a`. So I guess we need to start not just the device codename in the ini file, but also the arch. And this means templates are an even better idea, as each arch will need its own lockfile if we want this to be deterministic for more than one device :)

I'm not sure storing the specific versions in the ini is the right approach. Thinking just the app name, and then create a separate lockfile that's not really meant for people, but reads from during build. Maybe a lack of lockfile could mean you want the latest. So if someone wants to build with the latest, then just `rm whatever.lock` and build. I like lockfiles for managing the assembling of packages/APKs -- They're a pretty standard pattern.

File: README.md [hunk 5]
```diff
@@ -33,19 +33,15 @@ make build_deploy
...
+versions:
 
-This process relies on the `xpath` command in the `libxml-xpath-perl`
-package.
+This process relies on the `xpath` command in the `libxml-xpath-perl` package.
 
 ```
 sudo apt-get install libxml-xpath-perl
-make get_app_index
-make generate_applist_lockfile
```
Comment (patcon) [inline README.md]: resolving with xpath is pretty slow, and when I'm building a lot, I do miss the speediness. We can merge this as-is, but I'll later have to add a way to skip the xpath stuff, ideally generating the lockfile once in advance

File: settings.sample.ini (hunk 1 above)
Comment (SchnWalter) [inline settings.sample.ini]: Updated the download script, for now all apps are installed as user apps.

File: settings.sample.ini (hunk 3 above)
Comment (SchnWalter) [inline settings.sample.ini]: Shoud I put it in the [apps] group with a direct download link?

Is this the one? https://f-droid.org/repo/fennec-36.0.multi.android-arm.apk

File: settings.sample.ini (hunk 2 above)
Comment (SchnWalter) [inline settings.sample.ini]: Fixed, and i also updated the version code, code 1 is no longer in the F-Droid repository.

File: README.md (hunk 5 above)
Comment (SchnWalter) [inline README.md]: I thik we could do a better job in python. So I would live it like this for now if it works.

**Comment by patcon** [comment]
<3

**Pull request merged**